src/async.c

```diff
@@ -105,6 +105,18 @@ coap_free_async(coap_async_state_t *s) {
   }
 }
 
+void
+coap_delete_all_async(coap_context_t *context) {
+  coap_async_state_t *next = context->async_state;
+
+  while (next) {
```

Comment:
This loop should be replaced with `LL_FOREACH_SAFE`.

Reply:
Code corrected and pushed.